### PR TITLE
Ensure user is either an Admin or Manager before allowing submission reviews

### LINKED ISSUE #1866: Lockdown submission review functionality
### Problem description

Currently, submission reviews are enabled for users who have permission to change an XForm. We should lock down the ability to review data to only allow Admins (& Managers ?) to create, update, and edit submission reviews.

### Expected behavior

Only admins & managers should be able to review submissions.

### Additional Information

_Logs, [related issues](github.com/onaio/onadata/issues), weird / out of place occurrences, local settings, possible approach to solving this..._

### PR DESCRIPTION
**Pull request by DavisRayM**
### Changes / Features implemented

- Require Admin or Manager privileges in order to perform create, update & delete operations on `SubmissionReview` model.
- Updated tests

### Steps taken to verify this change does what is intended

- Updated tests
- [x] QA

### Side effects of implementing this change

- Editors will no longer be able to create, update or delete Submission Review objects

Closes #1866

### CODE REVIEW

**Comment by pld** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- onadata/apps/api/permissions.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ona/onadata/pullRequest?prid=5975563&bid=19593222)

File: onadata/apps/api/permissions.py [hunk 1]
```diff
@@ -163,6 +164,10 @@ class SubmissionReviewPermissions(XFormPermissions):
         'DELETE': ['logger.delete_xform'],
     }
 
+    def _check_is_admin_or_manager(self, user: User, xform: XForm):
```
Comment (pld) [inline onadata/apps/api/permissions.py:167]: Codacy found an issue: [Method could be a function](https://app.codacy.com/manual/Ona/onadata/pullRequest?prid=5975563)

**Comment by pld** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ona/onadata/pullRequest?prid=5975563&bid=19593222)

File: onadata/apps/api/permissions.py [hunk 2]
```diff
@@ -163,6 +164,11 @@ class SubmissionReviewPermissions(XFormPermissions):
         'DELETE': ['logger.delete_xform'],
     }
 
+    def _check_is_admin_or_manager(  # pylint: disable=no-self-use
+            self, user: User, xform: XForm) -> bool:
+        return OwnerRole.user_has_role(
+            user, xform) or ManagerRole.user_has_role(user, xform)
```
Comment (DavisRayM) [inline onadata/apps/api/permissions.py:170]: Should managers still be able to create, update & delete submission reviews?

Comment (ukanga) [reply onadata/apps/api/permissions.py:170]: Yes, I think so.

File: onadata/apps/api/tests/viewsets/test_submission_review_viewset.py [hunk 3]
```diff
@@ -196,8 +198,12 @@ def test_submission_review_permission(self):
         Test that submission review access to unauthorized users
         """
         data = self._create_submission_review()
+        form = Instance.objects.get(id=data['instance']).xform
         self._create_user_and_login('dave', '1234')
         extra = {'HTTP_AUTHORIZATION': 'Token %s' % self.user.auth_token}
+        # Editors should not be able to create, update, delete
+        # reviews. Only Admins and Managers should have these permissions
+        EditorRole.add(self.user, form)
```
Comment (ivermac) [inline onadata/apps/api/tests/viewsets/test_submission_review_viewset.py:206]: The irony of the role title and the `update` functionality 😄

**Pull request merged**